basics/ranges.md

```diff
@@ -0,0 +1,115 @@
+# Ranges
+
+Wenn der Compiler auf ein `foreach`-Konstrukt trifft...
+
+```
+foreach (element; range)
+{
+    // Loop body...
+}
+```
+
+wird es intern etwa wie folgt umgeschrieben:
+
+```
+for (auto __rangeCopy = range;
+     !__rangeCopy.empty;
+     __rangeCopy.popFront())
+ {
+    auto element = __rangeCopy.front;
+    // Loop body...
+}
+```
+
+Falls das Range-Objekt ein Referenztyp ist (z.B. `class`), wird es 
+verbraucht ist für weitere Iterationen nicht mehr verfügbar (es sei
+denn, der Schleifenrumpf bricht vor der letzten Iteration ab).
+Falls das Range-Objekt ein Werttyp ist, wird eine Kopie der Range 
```

Comment:
Du hast in einem PR Wert-Typ verwendet. Ich glaube es ist vor allem wichtig einheitlich zu sein.
Siehe auch: https://github.com/dlang-tour/german/issues/41